app/components/App/App.js

```diff
@@ -0,0 +1,180 @@
+    this.updatePathOnBoard = this.updatePathOnBoard.bind(this);
+    this.updateBoardColorState = this.updateBoardColorState.bind(this);
+  }
+
+  updateBoardColorState(updatedColor, newColor) {
+    this.setState(() => ({ updatedColor: newColor, }));
+  }
+
+  handleClickOnBoard(x, y) {
+    if (this.state.startButtonPressed) {
+      this.updateStartLocation(x, y);
+    } else if (this.state.endButtonPressed) {
+      this.updateEndLocation(x, y);
+    } else {
+      this.toggleAllowedStateOfCell(x, y);
+    }
+  }
+
+  toggleAllowedStateOfCell(x, y) {
+    const { board, } = this.state;
+    const boardCopy = copyBoard(board);
+    boardCopy[y][x].allowed = !this.state.board[y][x].allowed;
+    this.setState({ board: boardCopy, });
+  }
+
+  toggleStartSettingFlag() {
+    this.setState({ startButtonPressed: !this.state.startButtonPressed, });
+  }
+
+  toggleEndSettingFlag() {
+    this.setState({ endButtonPressed: !this.state.endButtonPressed, });
+  }
+
+  findAndDrawPath() {
+    const { startingPoint, endingPoint, board, } = this.state;
+    const boardCopy = copyBoard(board);
+    const path = findPath(boardCopy, startingPoint, endingPoint);
+    const stepCounter = -1;
+    this.drawPath(path, stepCounter);
+  }
+
+  drawPath(pathRemaining, stepNumber) {
+    if (pathRemaining.length === 0) {
+      return;
+    }
+    this.updatePathOnBoard(pathRemaining[0], stepNumber + 1);
+    window.setTimeout(this.drawPath.bind(this, pathRemaining.slice(1), stepNumber + 1), 500);
+  }
+
+  updateStartLocation(x, y) {
+    this.setState({
+      startingPoint: [ x, y ],
+      startButtonPressed: false,
+    });
+  }
+
+  updateEndLocation(x, y) {
+    this.setState(() => ({
+      endingPoint: [ x, y ],
+      endButtonPressed: false,
+    }));
+  }
+
+  updatePathOnBoard(step, stepNum) {
+    const { board, } = this.state;
+    const boardCopy = copyBoard(board);
+    const [ x, y ] = step;
+    boardCopy[y][x].stepVisited	= stepNum;
+    this.setState({ board: boardCopy, });
+  }
+  render() {
+    const {
+      handleClickOnBoard,
+      toggleStartSettingFlag,
+      toggleEndSettingFlag,
+      findAndDrawPath,
+    } = this;
+    const handleClicks = {
+      handleClickOnBoard,
+      toggleStartSettingFlag,
+      toggleEndSettingFlag,
+      findAndDrawPath,
+    };
+    const gameProps = {
+      state: this.state,
+      handleClicks,
+    };
+    return (
+      <BrowserRouter>
+        <div className="container">
+          <Switch>
+            <Route
+              exact
+              path="/"
+              render={ () => (
+                <Game { ...gameProps } />
+                ) }
+            />
+          </Switch>
+        </div>
+      </BrowserRouter>
+    );
+  }
+}
+
+App.defaultProps = {
+  dimensions: {
+    x: 12,
+    y: 12,
+  },
+  colors: {
+    A: 'red',
+    B: 'blue',
+    C: 'green',
+  },
+};
+
+App.propTypes = {
+  dimensions: PropTypes.object,
+  colors: PropTypes.object,
+};
+
+module.exports = App;
```

Comment:
Don't mix CJS and ESM. It's a hack which only works in non standards compliant module bundlers and will stop working as soon as module loaders enforce spec compliance (which is already the case for most of them).
Replace with `export default App`

Reply:
done